Salvage and Mining Rework Pt 1 - differentiating the two

mikeysaurus opened the pull request:
Salvage medic was taken out of this PR, and put into a seperate PR here #1527 

# Short description
This PR aims to expand on and improve the gameplay experiences for the Salvage and Mining sub-departments, while attempting to not also detract from the rest of the stations experience.
* New role - Adding the Salvage Lead (requires 5 hours of QM)
* Separating Mining + Salvage access out
* Alongside existing Salvage department doors, add new ones - Mining department doors, and Salvage/Mining combo doors
* Separate Mining into their own new ticket vendor
* Add the mining shuttle that Grapegifter made to spawn at round start
* Large lockers for salvage and mining!
* Recolors cargo shuttle console, mining shuttle console, and salv shuttle console
* New jumpsuit sprites for salvage and mining specialists

# Why we need to add this
Not recommending you read it (as its over 2,000 messages long 💀 ), but here was the originating Discord suggestion thread: https://discord.com/channels/1272545509562777621/1395148688703226040

## New Roles:

### Salvage Lead

Requires 5 hours of QM

**The Salvage Lead** is a role that is meant to solve a few problems:
* Problem: The QM does not know what Salvage is up to
  - Solution: They both have Expedition channel access to maintain tight communication with each other.
* Problem: Salvage not following SOP
  - Solution: There is now 1 responsible individual that can actually be held accountable, as the QM often could not realistically maintain control over salvage. The Lead is with the rest of the team almost always, so they _can_ keep a much tighter leash on the team.
* Problem: Salvage Specialists fight for authority. Too frequently would salvage be either stuck on station for over 15 minutes because no one can organize, _or_ a salvager picks a very difficult expedition and loads it immediately.
  - Solution: The Salvage Lead is the specialists' boss, and have a clear line of communication to report any troublemakers/problems to the QM.

The Salv Lead get the following items:
- Salvage Lead stamp (for stamping salvages reports to QM per SOP)
- Salvage Lead Hardsuit
- Salvage Decorative Mantle (the red pauldron) neck slot cosmetic
- Clipboard

## Access:
Salvage and Mining have been in a strange relationship, where mining specialists are not supposed to be trusted individuals given their lack of mindshield, yet they have built in access to an expedition-capable shuttle, the maxim hardsuit, and guns. 

This PR fixes that by establishing clear boundaries between salvage and mining:
* Salvage is a group of trusted individuals that are meant to be occasionally isolated, and in dangerous situations that require the usage of contraband that would otherwise make them an extreme threat to the station. They gather resources in big bursts from their expeditions, whether it is scrap/valuables from dungeons, or materials from strip-mining planets.
* Mining specialists on the other hand are meant to be steady resource gatherers, keeping the station supplied while salvage is out for 16+ minutes at a time. Often times though, Miners would join salvagers on expeditions and take advantage of the dangerous loot they were exposed to while also often being an antagonist.

With separate access, and their own ships now that are locked to their respective access types, the hope is that there will not be this unclear crossover between the roles that muddies further balancing attempts. Miners can no longer purchase weapons from their ticket vender.

## New Mining Shuttle

With the new access breakouts, Mining specialists have their own shuttle (that currently cannot do expeditions). This was a requested feature by many, and will further help alleviate the problems of mining specialists hitching rides on the salvage shuttle.

## Mining Shuttle + Salvage Shuttle Console Updates

The Mining Shuttle Console and Salvage Shuttle Console allow remote control of the respective salvage and mining shuttles. I cannot easily add access requirements on them, so my recommendation is to lock these in a restricted area (such as behind mining or salvage access doors) to mitigate griefing.

## Expedition Keys

Both the Salvage Lead and the QM have dedicated access to that channel via an encryption key in their headsets. The Captain also has the key in their locker, as opposed to being in their headset by default, as there is potential for the expedition channel to get quite spammy.

I did not want to give Expedition Keys out to circumvent the janky communication using expedition radios that I think makes Salvage unique, but to reinforce cohesion between the QM and the Salvage Lead, to combat the problems the server has had with rogue salvage crews that the QM has no idea of.

## Mining Ticket Vendor

To give the mining specialists some additional things to look forward to, they can now purchase handheld mass scanners and up to 5 each of the different PKA upgrade modules, ontop of most of the original salvage ticket vendor inventory (minus guns).

## Mining Suit Progression

Mining specialists roundstart with the hazards suit in their lockers, which is _fine_. From there, in their ticket vender, they can now purchase the "Mining Hardsuit" (which is a resprite of what was previously there - that version has been renamed to the Salvage Hardsuit for salvage to buy).

To follow a similar progression to salvagers, Miners can then buy their own version of the Maxim - the advanced mining hardsuit. This suit does not feature the combat benefits that the maxim does (lacking the same stat bonsues, and lacking the bubble shield), but instead features a built in jetpack to make the miner's jobs easier.

The goal of this was to still provide an incentive for the miners to actually do their job, without letting them get extremely valuable combat gear to use as an antag.


# Media (Video/Screenshots)
<img width="620" height="199" alt="Ajobequipment" src="https://github.com/user-attachments/assets/5048088f-cc4f-449f-b6da-154add5fdd0d" />

<img width="308" height="359" alt="Bnewdoors" src="https://github.com/user-attachments/assets/42d0482b-c44b-45fb-9218-cd2fd1360785" />
<img width="151" height="148" alt="Clockers" src="https://github.com/user-attachments/assets/2c5aed97-95df-46cd-97ed-46a7b79654e7" />
<img width="271" height="115" alt="Dmiscstuff" src="https://github.com/user-attachments/assets/2602bf97-9bb0-408d-880a-1b7e1b107e4d" />

<img width="125" height="60" alt="jobicons" src="https://github.com/user-attachments/assets/38b57100-6e15-425f-8d61-55d05cdb94e8" />

<img width="547" height="387" alt="grapes mining shuttle" src="https://github.com/user-attachments/assets/8937f788-ec16-49e0-be53-1dc2b5d429dc" />


<img width="451" height="380" alt="suitcomparisons" src="https://github.com/user-attachments/assets/3b68eea0-0b84-4b18-b6ff-03183715e88a" />



# Suggested Mapping Updates
- [ ] **Update all existing salvage doors to either dual-access salvage/mining doors, or separate out salvage and mining within Cargo**
- [ ] If able, expand the dock to accommodate having the current salvage shuttle, and the new mining shuttle
- [ ] Add the new mining ticket vender to the cargo department
- [ ] Add the new shuttle consoles (mining shuttle console, salvage shuttle console) behind the sub-department doors, to restrict who can access the consoles to just those jobs
- [ ] Add new Mining department signs (where relevant!)
- [ ] Update salvage locker rooms to include the new Salvage Lead lockers, and the new Large lockers for all of these roles
- [ ] Add Salvage Lead spawns to salvage locker room
- [ ] Make sure that the Mining department has a magnet available to them on station

# Checks

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**

:cl: mikeysaurus
- add: New Role - Salvage Lead. Guide your salvage specialists to glory! (Thanks Choccy, Bort, deltaVelocity, and itsMaitake for the idea/sprites!)
- add: New "Mining" access type, granted to mining specialists
- add: Added new Mining sub-department doors for mappers
- add: Added new Salvage+Mining sub-department combo doors for mappers
- add: Added new Mining wall poster, and directional signs for mappers
- tweak: Renamed the Wizden salv shuttle's black doors to include "retro" in filename
- remove: Removed mining specialist access from salvage ticket vendor
- add: Added new miner ticket vendor
- add: Salvagers and Miners can purchase Floodlights from their respective vendors
- add: Added Grape's mining shuttle. Mining specialists, report to the "Breaker" for assignment (Thanks Grape for the shuttle!)
- tweak: Miners can only access the Breaker mining shuttle, Salvage can only access the Reclaimer salvage shuttle.
- add: New jumpsuit sprites for Salvage and Mining Specialists (thanks itsMaitake for the sprites!)
- tweak: Renamed "mining hardsuit" to "salvage hardsuit"
- add: Added a new "mining hardsuit" for miners to purchase. It's an upgrade over their space hazards suit
- add: Added a new "advanced mining hardsuit" for miners to purchase as a maxim-like goal. Does not possess the same combat bonuses but does have a built-in jetpack!
- add: Adds mining shuttle console for remote access
- add: New colors for the cargo department shuttle consoles, so they don't all look the same!
- tweak: Mining and Salvage Shuttle Consoles require their department access types!
- fix: salvage shuttle console will now, when deconstructed, drop the correct circuitboard
- add: Gave QM, Salv Lead, and Captain expedition encryption keys. The radio channel keybind has been updated to ":o". It functions like the freelance channel, except cargo's Expedition Radios also communicate on it.
- add: Adds a "salvage encryption key", containing both supply and expedition channels. 1 given to QM's locker, otherwise intended for Admeme Salvage training events.
- add: Adds 2 Salvage/Security crates that spawn on the Reclaimer, intended for Salvage to safely store contra in while they transport it to Security.

mikeysaurus committed:
add salvage role draft

mikeysaurus committed:
add salvage role draft

mikeysaurus committed:
Merge branch 'ss14Starlight:Starlight' into salvageRolesExpansion

mikeysaurus committed:
mining ticket vendor: first draft

mikeysaurus committed:
mining ticket vendor: first draft

mikeysaurus committed:
Salvage Medic: add base files

mikeysaurus committed:
Rebase onto starlight-dev

mikeysaurus committed:
Salvage Lead: fix job file

mikeysaurus committed:
Salvage Lead: fix job file

mikeysaurus committed:
Salvage Lead + Medic: Add base for customization and lockers

mikeysaurus committed:
Salvage Lead + Medic: Add base for customization and lockers

mikeysaurus committed:
Mining Access: initial edits, still need to make door protos

mikeysaurus committed:
Mining Access: initial edits, still need to make door protos

mikeysaurus committed:
Salvage Lead + Medic: added to department description

mikeysaurus committed:
Salvage Lead + Medic: added to department description

Simpathey commented:
Ok hear me out. I love the idea of different roles and I have a great idea to add it with additional RP possibilities. I think heads of department should get "badge" items in their lockers at round start. They can give these to who they wish to designate leads like Head of anoms in science or salvage team lead on exped ext. This badge could modify the icon of the player slightly, maybe a 1 or tick mark in the corner of the icon. I was thinking of adding the badge system but don't want to conflict with your work. Let me know what you think. Adding the badge might be a clean addition I think. Also I agree that mining and salv need a bit more differentiation.

RHSvenson commented:
- Instead of salvage lead, call him Expedition Leader. I think it sounds cooler and is a lot more fitting.
- Similarly, I like Expedmedic or Expedition Medic a lot more

Otherwise, this is exactly what we need. A mining ticket vend too, etc. Good stuff IMO.

mikeysaurus commented:
> Ok hear me out. I love the idea of different roles and I have a great idea to add it with additional RP possibilities. I think heads of department should get "badge" items in their lockers at round start. They can give these to who they wish to designate leads like Head of anoms in science or salvage team lead on exped ext. This badge could modify the icon of the player slightly, maybe a 1 or tick mark in the corner of the icon. I was thinking of adding the badge system but don't want to conflict with your work. Let me know what you think. Adding the badge might be a clean addition I think. Also I agree that mining and salv need a bit more differentiation.

This is actually an idea that has come up before in the suggestion thread to add manager roles (which is where these role ideas come from). I'm personally in favor of such an idea!

Regardless though, I think the adding of two distinct roles to salvage is important.

mikeysaurus commented:
> * Instead of salvage lead, call him Expedition Leader. I think it sounds cooler and is a lot more fitting.
> * Similarly, I like Expedmedic or Expedition Medic a lot more
> 
> Otherwise, this is exactly what we need. A mining ticket vend too, etc. Good stuff IMO.

I agree that is sounds cooler, but I want the "salvage" roles to maintain the same naming. I considered a rename of salvage to "Pathfinder", similar to your "Expedition", but I feel like the name salvage has a lot attached to it.

I might revisit that, but I think I'm going to stick with the name convention as "Salvage" for now.

mikeysaurus committed:
update all of my TODOs to DOTS for easier tracking

mikeysaurus committed:
update all of my TODOs to DOTS for easier tracking

mikeysaurus committed:
Salvage Roles: icons for all three. Need to revisit

mikeysaurus committed:
Salvage Roles: icons for all three. Need to revisit

mikeysaurus committed:
Salvage Roles: add new role icons, adjust shaftminer to purple color

mikeysaurus committed:
Salvage Roles: add new role icons, adjust shaftminer to purple color

mikeysaurus committed:
Salvage Roles: Add new PDAs and IdCards

mikeysaurus committed:
Salvage Roles: Add new PDAs and IdCards

mikeysaurus committed:
Vendor: add mining vend, edit salv vend

mikeysaurus committed:
Vendor: add mining vend, edit salv vend

mikeysaurus committed:
Salvage Icon: Update salvage locker to have new job icon

mikeysaurus committed:
Salvage Icon: Update salvage locker to have new job icon

mikeysaurus committed:
Vendor: update available categories

mikeysaurus committed:
Vendor: update available categories

mikeysaurus committed:
Salvage Roles + Access: add new locker textures, and access notes

mikeysaurus committed:
Salvage Roles + Access: add new locker textures, and access notes

mikeysaurus committed:
Doors: added new door types for miners, and salvage/miner combo

mikeysaurus committed:
Doors: added new door types for miners, and salvage/miner combo

mikeysaurus committed:
Salvage Roles: rearranged file locations, woops

mikeysaurus committed:
Salvage Roles: rearranged file locations, woops

mikeysaurus committed:
fix lockers

mikeysaurus committed:
fix lockers

mikeysaurus committed:
fix merge conflict hopefully

mikeysaurus committed:
Merge branch 'starlight-dev' into salvageRolesExpansion

mikeysaurus committed:
Salvage Roles: update meta for lockers

mikeysaurus committed:
Salvage Roles: update meta for lockers

mikeysaurus committed:
Merge remote-tracking branch 'Starlight-myFork/salvageRolesExpansion' into salvageRolesExpansion

mikeysaurus committed:
todo for proper mining shuttle

mikeysaurus committed:
todo for proper mining shuttle

mikeysaurus commented:
note to self - reference this PR for adding a mining console for mining shuttle https://github.com/ss14Starlight/space-station-14/pull/1091

mikeysaurus committed:
Merge branch 'starlight-dev' into salvageRolesExpansion

mikeysaurus committed:
Merge remote-tracking branch 'Starlight-myFork/salvageRolesExpansion' into salvageRolesExpansion

mikeysaurus commented:
i rebased onto starlight-dev, hopefully didnt fuck everything up :)

mikeysaurus committed:
Mining Shuttle: rename + add dedicated spawn

mikeysaurus committed:
notes

mikeysaurus committed:
Radio: add encryption key

mikeysaurus committed:
Radio: sprite for key

mikeysaurus committed:
note

mikeysaurus committed:
Revert weird whitespace in "notes" commit

This reverts commit 6e6bf5c118fe294f4a917a801d42fc65e600c716.

mikeysaurus committed:
Radio: limit listen range

mikeysaurus committed:
Merge branch 'ss14Starlight:Starlight' into salvageRolesExpansion

mikeysaurus committed:
Salvage Roles: base spawns, jumpsuits,

mikeysaurus committed:
contraband note

mikeysaurus committed:
Mining Shuttle: add grape's mining shuttle

mikeysaurus committed:
Update access and lockerfills

mikeysaurus committed:
Salvage Roles: update jumpsuit sprites

mikeysaurus committed:
Access: Add mining/salv doors + access, rename mining shuttle black glass to retro

mikeysaurus committed:
Access: Add mining doors, salvage/mining combo doors, and remake black salv shuttle doors

mikeysaurus committed:
Access: Add mining doors, salvage/mining combo doors, and remake black salv shuttle doors

mikeysaurus committed:
fix merge

mikeysaurus committed:
Merge branch 'fixmerge' into salvageRolesExpansion

mikeysaurus committed:
this time for real, actually undo wizden renames

mikeysaurus committed:
minor fixes

mikeysaurus committed:
Vendor: inventories need separate categories. vendor now works

mikeysaurus committed:
add mining signs

mikeysaurus committed:
Merge branch 'ss14Starlight:Starlight' into salvageRolesExpansion

mikeysaurus committed:
Job fixes

mikeysaurus committed:
removing TODOs

mikeysaurus committed:
Salvage Roles: more locker stuff

mikeysaurus committed:
added pickaxe sign only

mikeysaurus commented on Resources/Prototypes/_StarLight/Roles/Jobs/Cargo/salvage_medic.yml:
```diff
@@ -0,0 +1,37 @@
+      role: JobSalvageSpecialist
+      time: 36000 # 10 hours
+    - !type:DepartmentTimeRequirement
+      department: Medical
```
To encourage the idea of the medic being a "paramedic" rather than a replacement for Salvage going to medical, should this require paramedic hours played, instead of just medical department hours?

CorruptOmega replied:
Maybe medical doctor and paramedic split? If they need to patch up in the field... Would think they need basic treatment skills and how to extract people to full medical.

mikeysaurus replied:
Yeah i made them paramedic, as that requires medical hours.

mikeysaurus committed:
Gave exped encryption key to Salvage Lead's roundstart headset ONLY

mikeysaurus committed:
Salvage Roles: more lockers!

mikeysaurus committed:
Salvage Roles: base for hardsuit

mikeysaurus committed:
organize mining specialist lockerfills

mikeysaurus committed:
Finalize large lockers yay

mikeysaurus committed:
for some reason i called lockers "equipment"

mikeysaurus committed:
unflipped the sprites

mikeysaurus committed:
add new mining hardsuit

mikeysaurus committed:
Merge branch 'ss14Starlight:Starlight' into salvageRolesExpansion

mikeysaurus committed:
Starlight - Updated airlocks for Salvage access ONLY

mikeysaurus committed:
Removed exped computer, added exterior signal buttons

mikeysaurus committed:
Salvage Lead rubber stamp for exped reports

mikeysaurus committed:
add seperate mining contra tags now that access is seperated

mikeysaurus committed:
rename miner's maxim, fixed jetpack bug

mikeysaurus committed:
itsmaitake versions of salv/mining jumpsuits and skirts. removed salvlead suits, will come back to it

mikeysaurus committed:
fixed meta

mikeysaurus committed:
remove salvlead/med suits, will do later

mikeysaurus committed:
Merge remote-tracking branch 'Starlight/Starlight' into salvageRolesExpansion

# Conflicts:
#	Resources/Locale/en-US/_Starlight/job/job-desc.ftl
#	Resources/Locale/en-US/_Starlight/job/job.ftl
#	Resources/Prototypes/_StarLight/Loadouts/loadout_groups.yml
#	Resources/Textures/_Starlight/Interface/Misc/job_icons.rsi/meta.json

mikeysaurus committed:
add mining shuttle remote control

mikeysaurus committed:
update locker fill for medic

mikeysaurus committed:
Salvage Roles: add salv lead and med suits and sprites.

mikeysaurus committed:
remove comment

mikeysaurus committed:
add salv lead/med jumpsuits and skirts

mikeysaurus committed:
add ceremonial salv mantle

mikeysaurus committed:
add vox sprites for salv lead and med

mikeysaurus committed:
almost done at last

mikeysaurus committed:
balance hardsuits

mikeysaurus committed:
Merge branch 'ss14Starlight:Starlight' into salvageRolesExpansion

mikeysaurus committed:
add dual encryption key for salv training events

mikeysaurus committed:
rename grapes shuttle file, access changes

mikeysaurus committed:
add encryption key box

mikeysaurus committed:
add treps salv shuttle

mikeysaurus committed:
spawn guaranteed fuel on salvage expeds

mikeysaurus committed:
update access

mikeysaurus committed:
nerf guaranteed fuel to 10

mikeysaurus committed:
Merge branch 'starlight-dev' into salvageRolesExpansion

mikeysaurus committed:
fix merge

mikeysaurus committed:
forgot these woops

mikeysaurus committed:
minor tweaks

mikeysaurus committed:
minor tweaks again

mikeysaurus commented on Resources/Prototypes/_StarLight/Entities/Clothing/Head/hardsuit-helmets.yml:
```diff
@@ -238,3 +238,119 @@
+    energy: 3
+  - type: Tag
+    tags:
+    - CorgiWearable
```
Note to self - Verify that these suits with "CorgiWearable" actually have corgi models
and also make sure the "ChameleonOutfit" tag actually works too

mikeysaurus replied:
i think we good now

mikeysaurus commented on Resources/Prototypes/_StarLight/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -353,4 +353,171 @@
+  description: A lightweight decorated hardsuit designed for industrial EVA in zero gravity.
+  components:
+    - type: Sprite
+      sprite: _Starlight/Clothing/OuterClothing/Hardsuits/leadspatio.rsi
```
pending new slightly modified sprites from itsMaitake - want to have a bit of color on the spatio suit when the mantle is not being worn

mikeysaurus replied:
done https://github.com/ss14Starlight/space-station-14/pull/1044/commits/60e7f027f83dcdb02bd10bd0cfcd53f6f1ed7fd2

APOCAMANDER commented:
what items does salvage medic get access to that regular salv does not? same question for leader.

mikeysaurus commented:
oops wrong button

> what items does salvage medic get access to that regular salv does not? same question for leader.

great question. I will elaborate on the PR description with way more information, Im realizing I neglected to mention that.
Salv Lead gets:
* Salvage Lead stamp (for stamping salvages reports to QM per SOP)
* Salvage Lead Hardsuit
* Salvage Decorative Mantle (the purple pauldron + waistcoat)
* Clipboard

Salv Med gets:
* Medical hud glasses
* Salvage Med Hardsuit
* Emergency Rollerbed
* Box of bodybags
* Medical + Cryogenics department access

mikeysaurus commented:
PR description has been updated, I hope things are a bit more clear now. Added new sections for other changes i made that I realized I neglected to fully elaborate on.

mikeysaurus commented on Resources/Prototypes/_StarLight/Entities/Structures/Machines/Computers/computers.yml:
```diff
@@ -171,8 +171,40 @@
+- type: entity
+  parent: BaseComputerShuttle
+  id: ComputerShuttleMining
+  name: mining shuttle console
```
There are three options for the Mining Shuttle and Salvage Shuttle remote consoles
1) Place them only in QMs room, so QM can remotely control the ship
2) Place them in Cargo, so anyone can control the ship (like the current Cargo shuttle console)
3) Place them in Cargo, but restrict access to the shuttle consoles to their access types only (Salvage and Mining)

I am in between 1 and 3, leaning more towards 3 personally. I am interested in feedback on this.

CorruptOmega replied:
3 would give easier access to everyone, even antags with additional accesses. So I would lean that way more.

mikeysaurus replied:
Agreed. I think locking them in the QMs room is too restrictive and boring, but giving everyone access will just lead to potential round removals by griefers etc

mikeysaurus replied:
i lied, shuttle console cant easily have access req tacked on. will deal with at a future time. Going to leave at Mappers discretion I suppose, although I will recommend placing consoles within the Mining/Salvage areas (where there is restricted access).

mikeysaurus committed:
updated reference to new name of adv mining suit

mikeysaurus committed:
maitakes sprites for base salv lead armor

mikeysaurus commented on Resources/Maps/_Starlight/Shuttles/cargo_salvage_pioneer.yml:
The current implementation of this salvage shuttle introduces the concept of needing to refuel the salvage shuttle. Onboard is a plasma generator, and a uranium generator. Starting fuel guarantees around ~40 minutes of fuel, and expeds are guaranteed to have 10 plasma and 10 uranium to prevent being stranded in space.

Curious on thoughts regarding this - Trep wanted to test out a fuel system and this is what we came up with in a short time frame.

mikeysaurus replied:
I am unsure if this change will get the result that is desired, which is salv forcing to come back to station. The question in my mind is do we want to still try it out, or do we want to shelve this fuel req until we can develop something more comprehensive?

mikeysaurus committed:
merge attempt 2

mikeysaurus committed:
merge actually works now bless

RHSvenson commented:
Huge amount of work, great job

mikeysaurus committed:
Merge remote-tracking branch 'Starlight/starlight-dev' into salvageRolesExpansion

# Conflicts:
#	Resources/Locale/en-US/_Starlight/objectives/conditions/steal-target-groups.ftl

mikeysaurus committed:
remove chameleon tag

mikeysaurus committed:
Merge branch 'ss14Starlight:Starlight' into salvageRolesExpansion

mikeysaurus committed:
Taking salvmed out, putting in different PR

mikeysaurus commented:
Salvage medic was taken out of this PR, and put into a seperate PR here https://github.com/ss14Starlight/space-station-14/pull/1527

mikeysaurus committed:
Merge remote-tracking branch 'Starlight/starlight-dev' into salvageRolesExpansion

# Conflicts:
#	Resources/Changelog/ChangelogStarlight.yml

mikeysaurus committed:
Merge remote-tracking branch 'Starlight/starlight-dev' into salvageRolesExpansion

# Conflicts:
#	Resources/Prototypes/_StarLight/Entities/Structures/Machines/Computers/computers.yml

mikeysaurus committed:
Added Salvage Lead access type for lead lockers

mikeysaurus committed:
access fixes

mikeysaurus committed:
add chameleon outfit

ShadesMars commented:
Bumping this up as il review it.

mikeysaurus committed:
New sprites per feedback

mikeysaurus commented:
@PubliclyExecutedPig Do you mind reviewing the two shuttles (both previously approved by you I think? But worth a second look) and maybe also my mapping recommendations noted in the PR description

mikeysaurus reviewed (commented):
@BORTBORTBORTBORTBORT I believe these versions of the sprites should align with the direction you wanted to take them in. Let me know if you want more changes!

Edit: stupid git didnt link the changes. here is the commit i am referencing https://github.com/ss14Starlight/space-station-14/pull/1044/commits/044c6e3e2c7a54c54e1f226a5964365af14389e1

mikeysaurus commented:
Failing test is unrelated to my changes.
This PR is ready for review

mikeysaurus committed:
Merge remote-tracking branch 'Starlight/starlight-dev' into salvageRolesExpansion

# Conflicts:
#	Resources/Changelog/ChangelogStarlight.yml
#	Resources/Prototypes/Access/misc.yml
#	Resources/Prototypes/StatusIcon/job.yml
#	Resources/Prototypes/_StarLight/Catalog/Fills/Boxes/general.yml
#	Resources/Prototypes/_StarLight/Entities/Clothing/Head/hardsuit-helmets.yml
#	Resources/Prototypes/_StarLight/Entities/Clothing/OuterClothing/hardsuits.yml
#	Resources/Prototypes/_StarLight/Store/categories.yml

mikeysaurus commented:
Made some corrections to the branch, I had some gross commits that were confusing things. Just re-tested and things are working as expected. I don't believe the failing test is something in this branch.

This PR is once again ready for review @ss14-Starlight @FoxxoTrystan <3

mikeysaurus committed:
Merge remote-tracking branch 'Starlight/starlight-dev' into salvageRolesExpansion

# Conflicts:
#	Resources/Prototypes/_StarLight/Entities/Clothing/Uniforms/jumpsuits.yml
#	Resources/Textures/_Starlight/Interface/Misc/job_icons.rsi/meta.json

mikeysaurus committed:
moving treps shuttle+guaranteed fuel to a different PR

mikeysaurus commented:
Removed the new salvage shuttle that Trep made, the Pioneer, and moved it to a new PR #1731

mikeysaurus commented:
@CawsForConcern I'd be curious what your thoughts are on the potential mapping updates required in this PR as well.

mikeysaurus committed:
add crate back

mikeysaurus committed:
Reverted Mining shuttle to wizden, SL has a _Starlight version now

mikeysaurus commented:
The large size is due to [last commit](https://github.com/ss14Starlight/space-station-14/pull/1044/commits/1147c9b766974ab875439ce9571e4eee0bff6ca3). I copied the starlight version of the salvage shuttle to a _Starlight directory, and reverted the Wizden mining.yml to be just a wizden shuttle.

CawsForConcern commented:
> @CawsForConcern I'd be curious what your thoughts are on the potential mapping updates required in this PR as well.

I think for a lot of maps, Salvage is going to remain as a mixed Salvage + Mining room usually due to lack of space, so I imagine that a lot of maps are just going to use the mixed access doors.

Mapping in stuff like the new ticket vendor or lockers shouldn't be a huge issue, it'll mostly just be a time thing.

The big ticket issue is that many Salvage docks do *not* have space for another shuttle to dock at. Some do, most don't. For the most part, this just means that some maps need locker docking arms so that they can park shuttles in parallel

Here's a quick MS Paint drawing of what I mean.
<img width="1036" height="674" alt="image" src="https://github.com/user-attachments/assets/89a32972-2c98-441e-bee8-d76c165a1277" />

fig. 1 - how some stations do docking

<img width="1034" height="682" alt="image" src="https://github.com/user-attachments/assets/f1a867f6-8fdf-407e-adbb-dc303f1ab418" />

fig. 2 - how this could be fixed on most stations, by making a docking arm that stations can park on either side of, doubling the docking capacity

In short:
- New role lockers is easy
- New role machines is easy
- New role spawnpoints is easy
- Dedicated Departmental spaces for Mining is hard, some stations are just really constrained for space and a lot of stations have them as 'shared' spaces
- Adjusting maps so both shuttles can dock is not hard, it will just take a lot of time to do.

mikeysaurus commented:
@CawsForConcern I agree on all of your points, I'm glad to know I'm not too far off. Most maps will probably need an extended docking arm for both shuttles, yeah. And as for the dedicated mining department space - I agree. I mostly wanted to add it to allow for future mapping potential! So totally fine if no dedicated areas make it in yet

mikeysaurus committed:
update my attribution

Conflee commented:
If these have the current pouldron sprites you showed me most recently (redish?) then all sprites approved. Just confirm that first.

mikeysaurus committed:
updates to mining stuff per feedback

mikeysaurus commented:
Thanks to some feedback on the discord, I have made these updates.
* added tank dispenser to mining shuttle (i still haven't figured out mapping gas stuff)
* added borg recharger to mining shuttle
* added mass scanner to mining shuttle
* added handheld mass scanner to mining ticket vendor (200 tickets)
* added PKA upgrades to mining ticket vendor (5 each, 300 tickets)

mikeysaurus committed:
Merge remote-tracking branch 'Starlight/starlight-dev' into salvageRolesExpansion

# Conflicts:
#	Resources/Prototypes/_StarLight/Entities/Clothing/Uniforms/jumpskirts.yml

mikeysaurus commented:
@Conflee 
<img width="69" height="108" alt="image" src="https://github.com/user-attachments/assets/9b336929-3e32-4267-babb-0503d2828357" />
Confirmed, the sprite you approved is in :)
